### PR TITLE
Lg-3175: Properly export Typography props and fix InlineCode prop definition

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

Exported typography interfaces now includ PolymorphicProps and InferredPolymorphicProps respectively

🎟 _Jira ticket:_ [Lg-3175](https://jira.mongodb.org/browse/Lg-3175)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: .changeset/heavy-lamps-hope.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/typography': patch
+---
+
+Exported typography interfaces now includ PolymorphicProps and InferredPolymorphicProps respectively
```
Comment (TheSonOfThomp) [inline .changeset/heavy-lamps-hope.md:5]: typo

**Review by TheSonOfThomp — COMMENTED** [review]
Should we be exporting base props?

File: .changeset/heavy-lamps-hope.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/typography': patch
```
Comment (TheSonOfThomp) [inline .changeset/heavy-lamps-hope.md:2]: Is this a minor change (new export)

File: .changeset/heavy-lamps-hope.md (hunk 1 above)
Comment (bruugey) [reply .changeset/heavy-lamps-hope.md:5]: 👍

File: packages/typography/src/Body/Body.types.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
 import { ResponsiveTypographyProps } from '../types';
 
 export type BodyFontWeight = 'regular' | 'medium';
-export type BodyProps = HTMLElementProps<'p'> &
-  ResponsiveTypographyProps & { weight?: BodyFontWeight };
+export type BaseBodyProps = ResponsiveTypographyProps & {
+  weight?: BodyFontWeight;
+};
+
+// For external consumption only
```
[the remark this answers is not in the document]
Comment (bruugey) [reply packages/typography/src/Body/Body.types.ts:10]: What type of note did you have in mind?

File: .changeset/heavy-lamps-hope.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bruugey) [reply .changeset/heavy-lamps-hope.md:5]: We're already exporting each component's interface - I think this is the right one to be referencing.

File: .changeset/heavy-lamps-hope.md (hunk 2 above)
Comment (bruugey) [reply .changeset/heavy-lamps-hope.md:2]: See above, we're already exporting

File: packages/typography/src/Link/Link.spec.tsx [hunk 4]
```diff
@@ -211,6 +211,46 @@ describe('packages/typography', () => {
...
+        };
+
+        // @ts-expect-error href is not allowed on explicit div
+        // eslint-disable-next-line
+        const asAndHref: LinkProps = {
+          as: 'div',
+          href: 'string',
+        };
+
+        // should expect error, but this does not
```
Comment (bruugey) [inline packages/typography/src/Link/Link.spec.tsx:234]: this should error @TheSonOfThomp, but it does not. I'm not sure that we should block the PR on this but i did want to call that out and see if you had any thoughts/ideas

Comment (TheSonOfThomp) [reply packages/typography/src/Link/Link.spec.tsx:234]: The prop type util isn't quite as robust as the component type check, We can add a ticket to fix this

**Review by TheSonOfThomp — APPROVED** [review]
👍

File: packages/typography/src/Link/Link.spec.tsx (hunk 4 above)
Comment (bruugey) [reply packages/typography/src/Link/Link.spec.tsx:234]: https://jira.mongodb.org/browse/LG-3227

**Pull request merged**